### ISSUE TITLE
manual vacation caused crash

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#313]
FWIW, 21 is `Troutboy `.

```
thread '<unnamed>' panicked at 'player PlayerId(21) not in got to do_advanced', mb2/src/table.rs:1857:13
stack backtrace:
  13:     0x56033c65260a - mb2::table::Table::do_advanced::hbad43dab53e23839
                               at mb2/src/table.rs:1857
  14:     0x56033c65260a - mb2::table::Table::player_action::h8e895345c1c7a5ab
                               at mb2/src/table.rs:1843
  15:     0x56033c6566f3 - <mb2::table::Table as mb2::table::UserActions>::vacation::h1f9a9244a79b754b
                               at mb2/src/table.rs:2797
```

I have not been able to reproduce this in demo mode, but I also haven't done a query of the database to see what Kevan had done because we started an `AREO` game as a substitute and I'm nervous that slow writing from `psql` to my buffer might tie up the db too long and cause trouble with our game.

My guess is that once I query the db to figure out everything that happened going in, I'll have something I can reproduce and fix, but I'm not going to tag this as `easy` until I do that.

**Comment by ctm** [comment]
The panic is due to:
```
    fn vacation(&mut self, seat: SeatIndex) {
...
        if self.action_or_discard_on(seat) {
            self.player_action(seat);
        }
    }
```
IOW, although Kevan had folded, `action_or_discard_on` was returning true. That leads back to our good friend `next_to_act`, which has been known to be incorrect at times "when it doesn't matter" (/me coughs). I just fixed an all-in bug related to that field, but I don't think this hand involved anyone all-in.

I dumped parts of the db last night to see if I could figure out how `next_to_act` could be out of phase, but I didn't see anything. I'll try again in the cold gray light of dawn...

**Comment by ctm** [comment]
This is the same bug as #314, which I've already fixed. What I overlooked last night is that there was indeed an all-in (specifically, `thumbers`). So, when `Troutboy` folded, there was no more action left, but `next_to_act` was never cleared (before the addition of the pause code, that didn`t matter) and so when he went on vacation, it detected an inconsistent state. It turns out, that inconsistent state _wouldn't_ have hurt anything here, but ... as we can see from #314 it was indeed a bad state to be in.

I'm going to rollback my fix and reproduce this locally and then verify that the problem doesn't happen with my fix, then close this ticket.

**Comment by ctm** [comment]
Sure enough, with the old code, I could reproduce the crash simply by starting a demo with 3 players, having the first two jam, then having the third fold and then manually go on vacation. Crash!

The code that I already deployed, that fixed #314 fixed this bug too.
